query/filter/key_builder.go

```diff
@@ -162,15 +179,140 @@ func (s *StrictEqKeyComposer) Compose(selectors []*Selector, userDefinedKeys []*
 					return nil, errors.InvalidArgument("OR is not supported with composite primary keys")
 				}
 
-				key, err := s.keyEncodingFunc(sel.Matcher.GetValue().AsInterface())
+				primaryKeyParts := s.buildIndexPartsFunc(sel.Field.Name(), sel.Field.DataType, sel.Matcher.GetValue().AsInterface())
+
+				eqkey, err := s.keyEncodingFunc(primaryKeyParts...)
 				if err != nil {
 					return nil, err
 				}
 
+				key := newKeyType(EQUAL, schema.UnknownType, []keys.Key{eqkey})
 				allKeys = append(allKeys, key)
 			}
 		}
 	}
 
 	return allKeys, nil
 }
+
+const (
+	RANGE     string = "range"
```

Comment:
is there a use case of setting Range/FullRange in KeyReadType?

Reply:
Yes. I haven't implemented that part. But we can use that to help with query planning. A `range` should be a smaller query than a `FULLRANGE` and a `EQUAL` would be the most narrow query.